### DISCUSSION TITLE
Enum serialization dynamic selection

### DISCUSSION THREAD

**Discussion by SadE54** [nlohmann/json#3980] [Q&A]
Hi,

Currently I have an application where I deserialize json with integer values for some enums. I want to switch to full string values  for enums, but with the possibility to still be able to parse integer value for enums, for compatibility.
Is it possible ? 
Or at least , to convert my json files , read with enum serialization disabled , and write back with it activated ?

Regards.

**Comment by nlohmann** [comment]
Can you provide an example?

**Comment by SadE54** [comment]
For example I have en enum like this :  
enum class PinType
{
    Bool = 0x01,
    Byte = 0x02,
    Word = 0x03,
    UWord = 0x04,
    Long = 0x05,
    ULong = 0x06,
    Float = 0x07
}

Until now serializing/deserializing was done using numerical value
In the JSON , I have "PinType": 1 ,2 ,...
Now I'm using 
NLOHMANN_JSON_SERIALIZE_ENUM(PinType, {
{PinType::Bool, "Boolean"},
{PinType::Byte, "Byte"},
{PinType::Float, "Float"},
...

to serialize as string . It's working well , but I lost compatibility with old Json file. What I want to do is to be still able to parse "PinType" : 1
if "PinType" : "Bool" is not available
